File: Data Structures & Algorithms/partition-labels/submission-0.cs
public class Solution {
    public List<int> PartitionLabels(string s) {
        List<int> res = new List<int>();
        Dictionary<char, int> lastSeen = new();
        for(int i = 0; i < s.Length; i++) {
            lastSeen[s[i]] = i;
        }
        int end = 0, size = 0;
        for(int i = 0; i < s.Length; i++) {
            end = Math.Max(end, lastSeen[s[i]]);
            size += 1;
            if(i == end) {
                res.Add(size);
                size = 0;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> seen;
        int n = s.size();
        int i = 0, j = 0;
        int res = 0;
        while(j < n) {
            while(j<n && seen.find(s[j]) == seen.end()) {
                seen.insert(s[j++]);
            }
            res = max(res, j - i);
            i++;j=i;
            seen.clear();
        }
        return res;
    }
};
